src/map.cpp

```diff
@@ -1931,6 +1931,13 @@ bool ServerMap::saveBlock(MapBlock *block)
 
 bool ServerMap::saveBlock(MapBlock *block, MapDatabase *db)
 {
+	// If the database is read-only, clear the modified flag without saving
+	static bool could_not_save = false;
```

Comment:
it should not be only static but static thread_local as if you open a new world which is writeable you will block it

Reply:
fixed